docs/how-to/fuzzing-dotnet-with-libfuzzer.md

```diff
@@ -0,0 +1,222 @@
+`LLVMFuzzerTestOneInput` which takes a pointer to a read-only buffer of bytes,
+and the length of said buffer.  ([Tutorial using libFuzzer in
+C](https://github.com/google/fuzzing/blob/master/tutorial/libFuzzerTutorial.md))
+
+With libfuzzer-dotnet, developers provide an application that within `Main` calls the method `Fuzzer.LibFuzzer.Run`, with a callback that passes a read only byte-stream their function of interest. 
+
+> NOTE: libfuzzer-dotnet only works on Linux at this time.
+
+TL/DR: check out our [libfuzzer-dotnet example](../../src/integration-tests/libfuzzer-dotnet/)
+
+## Issues using libfuzzer-dotnet in OneFuzz
+* The `libfuzzer_coverage` task does not support the coverage features used by libfuzzer-dotnet.  (Work item: [#536](https://github.com/microsoft/onefuzz/issues/536))
+* The `libfuzzer_crash_report` does not support extracting unique output during analysis, making the crash de-duplication and reporting ineffective. (Work item: [#538]https://github.com/microsoft/onefuzz/issues/538))
+
+As such, a libfuzzer-dotnet template is available, which only uses the `libfuzzer_fuzz` tasks.  As these issues are resolve, the template will be updated to include the additional tasks.
+
+## Example
+
+Let's fuzz the `Func` function of our example library named [problems](../../src/integration-tests/libfuzzer-dotnet/problems/).
+
+1. Make sure sharpfuzz and a recent version of clang are installed.  We'll need these later.
+
+  ```
+  dotnet tool install --global SharpFuzz.CommandLine
+  sudo apt-get install -y clang
+  ```
+
+2. We need to build an application that uses `Fuzzer.LibFuzzer.Run` that calls our function `Func`.  For this example, let's call this [wrapper](../../src/integration-tests/libfuzzer-dotnet/wrapper/) 
+
+  The [wrapper/wrapper.csproj](../../src/integration-tests/libfuzzer-dotnet/wrapper/wrapper.csproj) project file uses SharpFuzz 1.6.1 and refers to our [problems](../../src/integration-tests/libfuzzer-dotnet/problems/) library locally.
+  ```xml
+  <Project Sdk="Microsoft.NET.Sdk">
+    <ItemGroup>
+       <ProjectReference Include="..\problems\problems.csproj" />
+    </ItemGroup>
+    <ItemGroup>
+      <PackageReference Include="SharpFuzz" Version="1.6.1" />
+    </ItemGroup>
+    <PropertyGroup>
+      <OutputType>Exe</OutputType>
+      <TargetFramework>netcoreapp3.1</TargetFramework>
+    </PropertyGroup>
+  </Project>
+  ```
+  
+  For our example [problems](../../src/integration-tests/libfuzzer-dotnet/problems/) library, our callback for `Fuzzer.LibFuzzer.Run` is straight forwards.  `Func` already takes a `ReadOnlySpan<byte>`.  If your functions takes strings, this would be the place to convert the span of bytes to strings.
+  [wrapper/program.cs](../../src/integration-tests/libfuzzer-dotnet/wrapper/program.cs)
+  ```C#
+  using SharpFuzz;
+  namespace Wrapper {
+    public class Program {
+      public static void Main(string[] args) {
+        Fuzzer.LibFuzzer.Run(stream => { Problems.Problems.Func(stream); });
```

Comment:
```suggestion 
        Fuzzer.LibFuzzer.Run(Problems.Problems.Func);
```

Reply:
I would prefer to leave this as is, primarily to give a clear point to do data conversion from the span of bytes into something else.